iwyu_output.cc

```diff
@@ -1426,6 +1426,71 @@ void CalculateIwyuForFullUse(OneUse* use,
   }
 }
 
+// This function removes all OneIncludeOrForwardDeclareLine's a that are desired
+// but not present but transitively included by some
+// OneIncludedOrForwardDeclareLine b that is desired and present
+// and adds a's symbol counts to b.
+vector<OneIncludeOrForwardDeclareLine> PruneTransitivelyPresent(
+    vector<OneIncludeOrForwardDeclareLine> const& lines,
+    const IwyuPreprocessorInfo* preprocessor_info) {
+  // In the first step, we copy all lines that
+  // * are either no include lines,
+  // * or that are an include file and present,
+  // * or that are an include file, not present and not desired,
+  // * or that are an include file, not present, desired, and whose include_file
+  //   isn't transitively included by another line which is an include file
+  // into one range and the remaining ones into another range.
+  const auto pred = [&lines, preprocessor_info](
```

Comment:
Can you name `pred` after its purpose. It looks something like `is_transitively_present`, maybe...?

Reply:
Good suggestion!